pkg/virt-launcher/virtwrap/agent-poller/agent_poller.go

```diff
@@ -410,3 +374,120 @@ func executeAgentCommands(commands []AgentCommand, con cli.Connection, agentStor
 		}
 	}
 }
+
+func executeApiOperations(commands []AgentCommand, con cli.Connection, agentStore *AsyncAgentStore, domainName string) {
+	for _, command := range commands {
+		domain, err := con.LookupDomainByName(domainName)
+		if err != nil {
+			// skip the command on error, it is not vital
+			continue
+		}
+
+		switch command {
+		case GET_INTERFACES:
+			if guestInfo, err := domain.GetGuestInfo(libvirt.DOMAIN_GUEST_INFO_INTERFACES, 0); err == nil {
+				agentStore.Store(GET_INTERFACES, convertToInterfaces(guestInfo))
+			}
+		case GET_OSINFO:
+			if guestInfo, err := domain.GetGuestInfo(libvirt.DOMAIN_GUEST_INFO_OS, 0); err == nil {
+				agentStore.Store(GET_OSINFO, convertToOSInfo(guestInfo))
+			}
+		case GET_HOSTNAME:
+			if guestInfo, err := domain.GetGuestInfo(libvirt.DOMAIN_GUEST_INFO_HOSTNAME, 0); err == nil {
+				agentStore.Store(GET_HOSTNAME, guestInfo.Hostname)
+			}
+		case GET_TIMEZONE:
+			if guestInfo, err := domain.GetGuestInfo(libvirt.DOMAIN_GUEST_INFO_TIMEZONE, 0); err == nil {
+				agentStore.Store(GET_TIMEZONE, convertToTimezone(guestInfo))
+			}
+		case GET_USERS:
+			if guestInfo, err := domain.GetGuestInfo(libvirt.DOMAIN_GUEST_INFO_USERS, 0); err == nil {
+				agentStore.Store(GET_USERS, convertToUsers(guestInfo))
+			}
+		}
+	}
+}
+
+func convertToInterfaces(guestInfo *libvirt.DomainGuestInfo) []api.InterfaceStatus {
+	var interfaceStatuses []api.InterfaceStatus
+	if guestInfo.Interfaces != nil {
+		for _, netInterface := range guestInfo.Interfaces {
+			if netInterface.Name == "lo" {
+				continue
+			}
+
+			interfaceIP, interfaceIPs := convertToIPAddresses(netInterface.Addrs)
+			interfaceStatuses = append(interfaceStatuses, api.InterfaceStatus{
+				Mac:           netInterface.Hwaddr,
+				Ip:            interfaceIP,
+				IPs:           interfaceIPs,
+				InterfaceName: netInterface.Name,
+			})
+		}
+	}
+	return interfaceStatuses
+}
+
+func convertToIPAddresses(ipAddresses []libvirt.DomainGuestInfoIPAddress) (string, []string) {
+	interfaceIPs := []string{}
```

Comment:
use `nil` instead of empty slice: `var interfaceIPs []string`

Reply:
Sure thanks.